src/root.cpp

```diff
@@ -60,6 +55,11 @@ void Root::index(Context *c)
 
 void Root::item(Context *c, const QString &uuid)
 {
+    // Some variables for optional features
+    bool socialMediaButtons = c->config(QStringLiteral("social"), true).toBool();
```

Comment:
Reimplemente Controller's:

    virtual bool preFork(Application *app);

And initialize the option as member variables like:

    private:
    bool m_downloadButton;

Reply:
I found example usage of preFork() at Cloudlyst:

- https://github.com/cutelyst/Cloudlyst/blob/2cc1bbd1dcdf48d15916def7daaf75c4c5095d60/src/webdav.h#L82
- https://github.com/cutelyst/Cloudlyst/blob/b27c7a7b1621bc57bab0f51b1905efcf5f7510f0/src/webdav.cpp#L636

But the compilation doesn't succeed when I make changes according to this. And when I try to adapt the variables are not effective.

https://paste.ubuntu.com/p/TfB63tD6kK/